sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -659,10 +659,18 @@ Table getTable(
         BackOff backoff,
         Sleeper sleeper)
         throws IOException, InterruptedException {
+      TableReference updatedRef = ref.clone();
+      if (updatedRef.getProjectId() == null) {
+        BigQueryOptions bqOptions = options.as(BigQueryOptions.class);
+        updatedRef.setProjectId(
+            bqOptions.getBigQueryProject() == null
+                ? bqOptions.getProject()
+                : bqOptions.getBigQueryProject());
+      }
```

Comment:
why not make this override directly on `ref`?

Reply:
I don't want to change the value of an input. It would change the for that object everywhere, which could have unexpected side effects

Reply:
+1 on keeping input unchanged, thanks!